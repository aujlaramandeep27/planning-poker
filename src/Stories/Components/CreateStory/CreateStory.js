import React from 'react';
import {connect} from 'react-redux';
import {viewStories} from '../../../Navigation/route-actions';
import {Page} from '../../../Common/Page';
import {PostStory} from '../../Actions/StoriesActions';

export function CreateStory({goToStories, postStory, currentMeeting}) {
    return (
        <Page title='Create Story' onBack={() => goToStories(currentMeeting._id)}>
            <form onSubmit={ (event) => { event.preventDefault(); postStory(event.target.storyInputName.value.trim()); } }>
                <div className="uk-margin">
                    <div className="uk-inline uk-width-1-1">
                        <span className="uk-form-icon" data-uk-icon="icon: file-edit"/>
                        <input name="storyInputName" className="uk-input" placeholder="Name" type="text"/>
                    </div>
                    <button className="uk-button uk-button-primary uk-margin-small-top uk-width-1-1">
                        CREATE
                    </button>
                </div>
            </form>
        </Page>
    );
}

function mapStateToProps(state) {
    return {
        currentMeeting: state.currentMeeting
    };
}

function mapDispatchToProps(dispatch) {
    return {
        goToStories: (meetingId) => dispatch(viewStories(meetingId)),
        postStory: (storyName) => dispatch(PostStory(storyName))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(CreateStory);
