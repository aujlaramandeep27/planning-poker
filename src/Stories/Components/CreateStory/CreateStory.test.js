import {CreateStory} from "./CreateStory";

it('CreateStory renders correctly', () => {
    const props = {
        goToStories: jest.fn(),
        postStory: jest.fn()
    }
    expect(CreateStory(props)).toMatchSnapshot();
});