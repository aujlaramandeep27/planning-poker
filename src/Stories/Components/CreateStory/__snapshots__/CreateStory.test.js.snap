// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateStory renders correctly 1`] = `
<Page
  onBack={[Function]}
  title="Create Story"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="uk-margin"
    >
      <div
        className="uk-inline uk-width-1-1"
      >
        <span
          className="uk-form-icon"
          data-uk-icon="icon: file-edit"
        />
        <input
          className="uk-input"
          name="storyInputName"
          placeholder="Name"
          type="text"
        />
      </div>
      <button
        className="uk-button uk-button-primary uk-margin-small-top uk-width-1-1"
      >
        CREATE
      </button>
    </div>
  </form>
</Page>
`;
